src/clarity-angular/popover/popover.ts

```diff
@@ -45,33 +56,203 @@ export class Popover {
         // Browsers don't agree with what to do if some of these are not specified, so we set them all to be safe.
         element.style.position = POSITION_ABSOLUTE;
         element.style.top = 0;
+        element.style.bottom = "auto";
         element.style.left = 0;
+        element.style.right = "auto";
     }
 
-    public anchor(anchor: any, direction: Direction,
-                  {offsetX = 0, offsetY = 0, userAnchorParent = false}: PopoverOptions = {}) {
+    public anchor(
+        anchor: any, anchorAlign: Point, popoverAlign: Point,
+        {offsetX = 0, offsetY = 0, useAnchorParent = false}: PopoverOptions = {}) {
+
         // TODO: we are assuming here that the popover is inside or next to the anchor.
         // We'd need to go up the popover tree too otherwise
+
         this.preventScrolling(anchor);
-        if (userAnchorParent) {
+        if (useAnchorParent) {
             anchor = anchor.parentNode;
         }
+        // explicitly override anchor's style to static
+        anchor.style.position = "static";
+
+        let marginLeft = parseInt(getComputedStyle(this.element).marginLeft, 10);
+        let marginRight = parseInt(getComputedStyle(this.element).marginRight, 10);
+        let marginTop = parseInt(getComputedStyle(this.element).marginTop, 10);
+        let marginBottom = parseInt(getComputedStyle(this.element).marginBottom, 10);
+
```

Comment:
Hmm `element` is already a private variable of the class, so it seems weird to assign that to yet another local variable. As per your suggestion, I created a variable so that `getComputedStyle(this.element)` is called only once. 

I will leave the `anchorRect.*`, and `popoverRect.*` as is. Creating additional variables like:
```
let anchorRectLeft = anchorRec.left;
```
doesn't change the refactorability too much. And because the `anchor()` function goes through a series of switch statements, the look up for `anchorRect.*` / `popoverRect.*` is only happening once each per `anchor()` function call.